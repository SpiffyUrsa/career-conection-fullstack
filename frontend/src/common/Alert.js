import React from "react";

function Alert({ messages }){
  
  const messageList = messages ?  messages.map(message =>
    <p className="alert alert-danger" key={message}>
      {message.slice(message.indexOf('.')+1)}
    </p>) : null

  return messageList
}

export default Alert