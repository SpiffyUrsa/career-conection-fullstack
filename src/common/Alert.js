import React from 'react'

/**
 * Routes --> {SignupForm, LoginForm, ProfileForm}
 * --> Alert
 * 
 * State
 * 
 * Props
 * 
 */
function Alert(){
  return(
    <div className='Alert'>
      <h1>This is an alert</h1>
    </div>
  )
}

export default Alert