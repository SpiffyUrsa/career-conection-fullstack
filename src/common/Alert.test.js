import React from "react";
import { render } from "@testing-library/react";
import Alert from "./Alert";

it("renders without crashing", function() {
  render(<Alert />);
});

it("matches snapshot for danger", function() {
  let errorMessages = ["instance.Everything is broken", "instance.Run for the hills"];
  const { asFragment } = render(<Alert messages={errorMessages} />);
  expect(asFragment()).toMatchSnapshot();
});