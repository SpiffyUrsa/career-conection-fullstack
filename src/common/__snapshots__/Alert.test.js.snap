// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for danger 1`] = `
<DocumentFragment>
  <p
    class="alert alert-danger"
  >
    Everything is broken
  </p>
  <p
    class="alert alert-danger"
  >
    Run for the hills
  </p>
</DocumentFragment>
`;
